src/passes/OptimizeInstructions.cpp

```diff
@@ -826,7 +826,7 @@ struct OptimizeInstructions : public WalkerPass<PostWalker<OptimizeInstructions,
         } else if (auto* ext = Properties::getSignExtValue(binary)) {
           // if sign extending the exact bit size we store, we can skip the extension
           // if extending something bigger, then we just alter bits we don't save anyhow
-          if (Properties::getSignExtBits(binary) >= store->bytes * 8) {
+          if (Properties::getSignExtBits(binary) >= uint32_t(store->bytes * 8)) {
```

Comment:
hmm, what was the warning here? (both are unsigned already, so it wasn't that I don't think?)

The cast should be to `Index` (what the function on the left returns, and which represents the proper index size in wasm modules - 32 bit for now).

Reply:
The warning was `C4018: ">=": signed/unsigned mismatch`, presumably due to integer promotion on the rhs